### PR TITLE
FOGL-8401: Fixed create control script with an empty data in step values and order key fix 

### PR DESCRIPTION
**Pull request by mshariq-nerd**

### CODE REVIEW

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-step.component.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export class AddStepComponent implements OnInit {
       const maxOrder = Math.max(...this.stepControlsList.map(o => o.order));
       this.initStepFormGroup(maxOrder + 1)
     } else {
-      this.initStepFormGroup(0)
+      this.initStepFormGroup(1)
```
Comment (praveen-garg) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-step.component.ts:66]: Please add inline comment here. What are we doing?

File: src/app/components/core/control-dispatcher/add-control-script/add-control-script.component.ts [hunk 2]
```diff
@@ -190,15 +191,25 @@ export class AddControlScriptComponent implements OnInit {
...
+            };
+          }).reduce((r, c) => ({ ...r, ...c }), {});
 
-        if ('write' in val) {
-          val['write'].values = values;
-        } else if ('operation' in val) {
-          val['operation'].parameters = values;
-        } else if ('script' in val) {
-          val['script'].parameters = values;
+        for (const key in val) {
```
Comment (praveen-garg) [inline src/app/components/core/control-dispatcher/add-control-script/add-control-script.component.ts:199]:
```
for (const key in val) {
  if (!['operation', 'script', 'write'].includes(key)) {
    console.log("Invalid key in ...");
    return;
  }
  
  if ( key === 'write') {
    val[key].values = values;
    if (Object.keys(values).length == 0) {
      delete val[key]?.values;
    }
    return;
  } 
  
  val[key].parameters = values;
  if (Object.keys(values).length == 0) {
    delete val[key]?.parameters;
  }
  
}
```

OR, if val can have any key and more than one;

```
for (const key in val) {
  if (!['operation', 'script', 'write'].includes(key)) {
    console.log("Invalid key in ...");
    continue;
  }
  
  if ( key === 'write') {
    val[key].values = values;
    if (Object.keys(values).length == 0) {
      delete val[key]?.values;
    }
  } 
  else {
    val[key].parameters = values;
    if (Object.keys(values).length == 0) {
      delete val[key]?.parameters;
    }
  }
}
```

File: src/app/components/core/control-dispatcher/add-control-script/add-control-script.component.ts [hunk 3]
```diff
@@ -190,15 +191,25 @@ export class AddControlScriptComponent implements OnInit {
...
+          if (['operation', 'script', 'write'].includes(key)) {
+            if (key !== 'write') {
+              val[key].parameters = values;
+              if (Object.keys(values).length == 0) {
+                delete val[key]?.parameters;
+              }
+              return;
```
Comment (praveen-garg) [inline src/app/components/core/control-dispatcher/add-control-script/add-control-script.component.ts:206]: Are we sure there will be only one of the ['operation', 'script', 'write'] in loop?

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-script/add-script.component.css [hunk 4]
```diff
@@ -5,3 +5,7 @@
 .field-label {
   text-align: left;
 }
+
+.required-field {
+  border-color: red;
+}
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-script/add-script.component.css:11]: It is duplicate here. It is also available in global css, so please remove.

**Review by monika-Dianomic — APPROVED** [review]

**Review by praveen-garg — APPROVED** [review]

**Review by hemantgarg923 — APPROVED** [review]

**Pull request merged**